Remove NSURLDownloadSPI.h

emw-apple opened the pull request:
#### 35683740872fdff569e5b4f2ceb010b3838c7087
<pre>
Remove NSURLDownloadSPI.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=274675">https://bugs.webkit.org/show_bug.cgi?id=274675</a>
<a href="https://rdar.apple.com/problem/128695608">rdar://problem/128695608</a>

Reviewed by Alexey Proskuryakov.

It was marked as a private header due to a single include in
WebDownload.h. It&apos;s weird to have an SPI header installed to the SDK, so
instead, just inline the declarations we need into WebDownload.h.

Additionally, import NSURLDownload.h when building for Mac Catalyst. The
preprocessor check was incorrectly assuming that it didn&apos;t have the
header available, because Catalyst is a TARGET_OS_IPHONE target despite
using the macOS SDK.

* Source/WebKit/MigratedHeaders-input.xcfilelist:
* Source/WebKit/MigratedHeaders-output.xcfilelist:
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKitLegacy/PlatformMac.cmake:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/Misc/NSURLDownloadSPI.h: Removed.
* Source/WebKitLegacy/mac/Misc/WebDownload.h: Inline the NSURLDownload
  declarations needed in public embedded builds. Declare NSURLDownload
  as visibility(&quot;hidden&quot;) to satisfy TAPI.

Canonical link: <a href="https://commits.webkit.org/279303@main">https://commits.webkit.org/279303@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e86a5327db2bae2bcd636ed5f1dbb9566a6b6fca

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53067 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32404 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5554 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56346 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3790 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/39268 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3517 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/43036 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2448 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55165 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/30165 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45807 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/24164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/27173 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3129 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1949 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49025 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3287 "Found 1 new test failure: media/video-unmuted-after-play-holds-sleep-assertion.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57941 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28208 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3244 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/50434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/29428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46027 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/49735 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/11579 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/30347 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29182 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Nice. I assume that you've done some production builds to verify this.

emw-apple committed:
Remove NSURLDownloadSPI.h
https://bugs.webkit.org/show_bug.cgi?id=274675
rdar://problem/128695608

Reviewed by Alexey Proskuryakov.

It was marked as a private header due to a single include in
WebDownload.h. It's weird to have an SPI header installed to the SDK, so
instead, just inline the declarations we need into WebDownload.h.

Additionally, import NSURLDownload.h when building for Mac Catalyst. The
preprocessor check was incorrectly assuming that it didn't have the
header available, because Catalyst is a TARGET_OS_IPHONE target despite
using the macOS SDK.

* Source/WebKit/MigratedHeaders-input.xcfilelist:
* Source/WebKit/MigratedHeaders-output.xcfilelist:
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKitLegacy/PlatformMac.cmake:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/Misc/NSURLDownloadSPI.h: Removed.
* Source/WebKitLegacy/mac/Misc/WebDownload.h: Inline the NSURLDownload
  declarations needed in public embedded builds. Declare NSURLDownload
  as visibility("hidden") to satisfy TAPI.

Canonical link: https://commits.webkit.org/279303@main
